### PR TITLE
Apply Remove overridden assignment JDT clean in pde.core

### PR DESCRIPTION
**Pull request by vogella**
Dogfooding the improve JDT cleanup from Bug 572440 to ensure the JDT
cleanup is stable and functional. Push in small chunks to make review
easier.

### CODE REVIEW

**Comment by akurtakov** [comment]
This commit can't be rebased automatically, please rebase manually.

File: ui/org.eclipse.pde.launching/src/org/eclipse/pde/launching/EclipseApplicationLaunchConfiguration.java [hunk 1]
```diff
@@ -99,10 +99,9 @@ public String[] getProgramArguments(ILaunchConfiguration configuration) throws C
 			programArgs.add(fWorkspaceLocation);
 		}
 
-		boolean showSplash = true;
 		String productID = LaunchConfigurationHelper.getProductID(configuration);
 		Properties prop = LaunchConfigurationHelper.createConfigIniFile(configuration, productID, fAllBundles, fModels, getConfigDir(configuration));
-		showSplash = prop.containsKey("osgi.splashPath") || prop.containsKey("splashLocation"); //$NON-NLS-1$ //$NON-NLS-2$
```
Comment (akurtakov) [inline ui/org.eclipse.pde.launching/src/org/eclipse/pde/launching/EclipseApplicationLaunchConfiguration.java:105]: You should keep the NON-NLS*

File: ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/editor/contentassist/XMLCompletionProposal.java [hunk 2]
```diff
@@ -329,17 +329,12 @@ private void modifyModel(IDocument document) {
 	 */
 	private void findExtensionVirtualPointValue(IPluginBase base) {
 
-		IDocumentRange range = null;
 		PDESourcePage page = fProcessor.getSourcePage();
 		// Ensure page is defined
 		if (page == null) {
 			return;
 		}
-		// When we inserted the extension element and extension point attribute
```
Comment (HannesWell) [inline ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/editor/contentassist/XMLCompletionProposal.java:338]: Should we preserve this comment?

**Review by HannesWell — COMMENTED** [review]
A really welcome clean-up! Just one remark.

**Comment by vogella** [comment]
Thanks for the review. I actually want to have separate PR for each plug-in to make review and updates easier. I will later see what I did wrong and also incooperate your feedback in the changes

File: ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/editor/contentassist/XMLCompletionProposal.java (hunk 2 above)
Comment (vogella) [reply ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/editor/contentassist/XMLCompletionProposal.java:338]: See https://github.com/eclipse-pde/eclipse.pde/pull/98

**Comment by vogella** [comment]
Ok, this PR only contains the pde.core changes, as indent. The feedback of @HannesWell and @akurtakov was incooperatated into the new PR

**Pull request merged**